spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/util/RequestContentDataExtractor.java

```diff
@@ -110,4 +125,29 @@
 
 		return result;
 	}
+
+	static Map<String, List<String>> findQueryParamsGroupedByName(HttpServletRequest request) {
+		Map<String, List<String>> result = new HashMap<>();
```

Comment:
Using something like this might simplify this code a bit 
`UriComponentsBuilder.fromUriString(request.getRequestURI()).build().getQueryParams();`